static/app/components/events/interfaces/performance/index.tsx

```diff
@@ -2,55 +2,91 @@ import styled from '@emotion/styled';
 
 import EventDataSection from 'sentry/components/events/eventDataSection';
 import {t} from 'sentry/locale';
-import space from 'sentry/styles/space';
-import {Event, EventTransaction, KeyValueListData, Organization} from 'sentry/types';
+import {EntryType, EventTransaction, KeyValueListData, Organization} from 'sentry/types';
 
 import KeyValueList from '../keyValueList';
 import TraceView from '../spans/traceView';
+import {RawSpanType, SpanEntry} from '../spans/types';
 import WaterfallModel from '../spans/waterfallModel';
 
 export type SpanEvidence = {
   parentSpan: string;
   repeatingSpan: string;
   sourceSpan: string;
-  transaction: string;
 };
 
 interface Props {
-  affectedSpanIds: string[];
-  event: Event;
+  event: EventTransaction;
   organization: Organization;
-  spanEvidence: SpanEvidence;
 }
 
-export function SpanEvidenceSection({
-  spanEvidence,
-  event,
-  organization,
-  affectedSpanIds,
-}: Props) {
-  const {transaction, parentSpan, sourceSpan, repeatingSpan} = spanEvidence;
+export function SpanEvidenceSection({event, organization}: Props) {
+  if (!event.perfProblem) {
+    return null;
+  }
+
+  const {
+    cause_span_ids: causes,
+    offender_span_ids: offenders,
+    parent_span_ids: parents,
+  } = event.perfProblem;
+
+  // For now, it is safe to assume that there is only one cause and parent span for N+1 issues
+  const sourceSpanId = causes[0];
+  const parentSpanId = parents[0];
+  const repeatingSpanIdSet = new Set(offenders);
+
+  // Let's dive into the event to pick off the span evidence data by using the IDs we know
+  const spanEntry = event.entries.find((entry: SpanEntry | any): entry is SpanEntry => {
+    return entry.type === EntryType.SPANS;
+  });
+  const spans: Array<RawSpanType> = spanEntry?.data ?? [];
+
+  const spanEvidence = spans.reduce(
+    (acc: SpanEvidence & {affectedSpanIds: string[]}, span) => {
+      if (span.span_id === sourceSpanId) {
+        acc.sourceSpan = span.description ?? '';
+      }
+
+      if (span.span_id === parentSpanId) {
+        acc.parentSpan = span.description ?? '';
+      }
+
+      if (repeatingSpanIdSet.has(span.span_id)) {
+        acc.repeatingSpan = span.description ?? '';
+        acc.affectedSpanIds.push(span.span_id);
+      }
+
+      return acc;
+    },
+    {
+      parentSpan: '',
+      sourceSpan: '',
+      repeatingSpan: '',
+      affectedSpanIds: [],
+    }
+  );
```

Comment:
This section is pretty complicated. There's a lot of renaming (from `offender_span_ids` to `offenders`, and then from that to `repeatingSpanIdSet` and from that to `affectedSpanIds`) and a complicated `reduce`.

It would be clearer (or at least a lot less code) to do something like this:

```js
import keyBy from 'lodash/keyBy';

...

const spansById = keyBy(spans, 'span_id');

const parentSpan = spansById[event.perfProblem.parentSpanIds[0]];
const causeSpan = spansById[event.perfProblem.causeSpanIds[0]];
const firstOffenderSpan = spansById[event.perfProblem.offenderSpanIds[0]];
```

Note: They keys in the response are actually `camelCase` as of yesterday!

and then lower

```js
value: parentSpan.description ?? '';
```

Reply:
You're right, I've spent too much time in Alberto's code and picked up the habit of using `reduce` to do everything 😂 

I can clean this up so it's more readable for sure!

Reply:
And I didn't know about `keyBy`, that's pretty :neat: and perfect for this usecase